pkg/messaging/mqtt/subscriber.go

```diff
@@ -18,14 +18,19 @@ import (
 var (
 	errSubscribeTimeout   = errors.New("failed to subscribe due to timeout reached")
 	errUnsubscribeTimeout = errors.New("failed to unsubscribe due to timeout reached")
+	errAlreadySubscribed  = errors.New("already subscribed to topic")
+	errNotSubscribed      = errors.New("not subscribed")
+	errEmptyTopic         = errors.New("empty topic")
+	errEmptyID            = errors.New("empty ID")
 )
 
 var _ messaging.Subscriber = (*subscriber)(nil)
 
 type subscriber struct {
-	client  mqtt.Client
-	timeout time.Duration
-	logger  log.Logger
+	client        mqtt.Client
+	timeout       time.Duration
+	logger        log.Logger
+	subscriptions map[string]map[string]mqtt.Token
```

Comment:
Unlike the CoAP token that identifies a session, `matt.Token` is, per documentation:
> Token defines the interface for the tokens used to indicate when actions have completed.

That is, the token is an artificial implementation of async operations within the MQTT library we use. However, if we want to have a correct implementation, we'll need to have an MQTT client (form the Paho lib) per ID (instead of mqtt.Token) - because MQTT client ID is unique per MQTT 3.1.1 spec.
Something like: `map[string]mqtt.Client` Where keys are client IDs, and values are MQTT Clients. Those MQTT clients will take care of its subscriptions per topic.